relay-server/src/actors/project.rs

```diff
@@ -685,13 +690,14 @@ impl Project {
             Ok(self.rate_limits.check_with_quotas(quotas, item_scoping))
         });
 
-        let rate_limits = envelope_limiter.enforce(&mut envelope, scoping)?;
+        let rate_limits = envelope_limiter.enforce(&mut envelope, scoping, |outcome| {
+            self.outcome_producer.do_send(outcome)
+        })?;
```

Comment:
I'd like to be able to just pass `self.outcome_producer` to `enforce` rather than this awkward closure. I did it this way to accommodate the `EnvelopeLimiter` unit tests, so that we can use a simple closure as a mock in place of the outcome producer. Let me know if you can suggest a better way to mock out the outcome producer.

The goal of injecting the outcome producer into `enforce` is to obviate the `RateLimitEnforcement` struct, as in d4af165760091cce83af67c3a5a3a7c905cb597c. I'm not certain it's worth it. If it isn't, we can just revert that commit.